ftp/src/main/scala/akka/stream/alpakka/ftp/javadsl/FtpApi.scala

```diff
@@ -91,7 +91,25 @@ sealed trait FtpApi[FtpClient] { _: FtpSourceFactory[FtpClient] =>
       basePath: String,
       connectionSettings: S
   ): Source[FtpFile, NotUsed] =
-    ScalaSource.fromGraph(createBrowserGraph(basePath, connectionSettings)).asJava
+    ScalaSource.fromGraph(createBrowserGraph(basePath, connectionSettings, f => true)).asJava
+
+  /**
+   * Java API: creates a [[Source]] of [[FtpFile]]s from a base path.
+   *
+   * @param basePath Base path from which traverse the remote file server
+   * @param connectionSettings connection settings
+   * @param branchSelector a function for pruning the tree. Takes a remote folder and return true
+   *                       if you want to enter that remote folder.
+   *                       Default behaviour is full recursiv which is equivalent with calling this function
+   *                       with [[ls(basePath,connectionSettings,f->true)]].
+   *
+   *                       Calling [[ls(basePath,connectionSettings,f->false)]] will emit only the files and folder in
+   *                       non-recursive fashion
+   *
+   * @return A [[Source]] of [[FtpFile]]s
+   */
+  def ls(basePath: String, connectionSettings: S, branchSelector: FtpFile => Boolean): Source[FtpFile, NotUsed] =
```

Comment:
Scala function for Java is only possible for Scala 2.12 but we support both 2.11 and 2.12 so this has to accept a `java.util.function.Function`

Reply:
thanks for the feedback .... I will update this and resubmit